include/nighthawk/common/request_source_plugin_config_factory.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+struct RequestSourceContext {
```

Comment:
Getting rid of this context because it does not make sense with the RegisterFactory mechanism we are using, since the actual arguments that are being passed are chosen before we decide which factory is being used.

Reply:
Yes, thanks for explaining, makes sense.